# ===== クラスEMCL =====
# フラグ
flag_init_noise: true # 初期位置にノイズを加えるか
flag_broadcast: true  # tf情報を含むbagファイルを使うときはfalse
is_visible: true      # パーティクルクラウドを表示（パブリッシュ）するか

# 基本設定
hz: 10              # ループ周期 [Hz]
particle_num: 420   # パーティクルの個数 [-]
move_dist_th: 0.025 # ロボットの移動開始判断用（スタートからの距離の閾値）[m]

# 初期ポーズ関連
init_x: 0.0        # 初期位置 [m]
init_y: 0.0        # 初期位置 [m]
init_yaw: 0.0      # 初期姿勢 [rad]
init_x_dev: 0.50   # 初期位置xの標準偏差 [m]
init_y_dev: 0.65   # 初期位置yの標準偏差 [m]
init_yaw_dev: 0.5  # 初期姿勢の標準偏差 [rad]

# リセット関連
alpha_th: 0.0017        # リセットに関する平均尤度の閾値 [-]
reset_count_limit: 5    # 連続リセットの回数の上限 [-]
expansion_x_dev: 0.05   # 膨張リセットの位置xの標準偏差 [m]
expansion_y_dev: 0.05   # 膨張リセットの位置yの標準偏差 [m]
expansion_yaw_dev: 0.01 # 膨張リセットの姿勢の標準偏差 [rad]

# センサ関連
laser_step: 10           # 何本ずつレーザを見るか(全1080本)[-]
sensor_noise_ratio: 0.03 # 距離に対するセンサノイズ比 [-]

# --- 柱に関する角度範囲の配列 [rad] ---
# 以下が無視される
# [0] < angle < [1], [2] < angle
# [num] -> num:index
ignore_angle_range_list: [rad(1.5*pi/16.0), rad(5.0*pi/16.0), rad(10.0*pi/16.0)]


# ===== クラスOdomModel =====
ff: 0.17   # 直進1[m]で生じる道のりのばらつきの標準偏差 [m]
fr: 0.0005 # 回転1[rad]で生じる道のりのばらつきの標準偏差 [m]
rf: 0.13   # 直進1[m]で生じる向きのばらつきの標準偏差 [rad]
rr: 0.2    # 回転1[rad]で生じる向きのばらつきの標準偏差 [rad]
